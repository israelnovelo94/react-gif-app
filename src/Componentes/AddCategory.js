import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {
  const [InputValue, setInputValue] = useState(''); //Enviamos string vacio para evitar el undefined y un error
  const handleInput = (e) => {
    setInputValue(e.target.value); //Actualizamos el valor del input
  }
  
  const handleSubmit = (e) => {
    e.preventDefault(); //prevenimos el refresh en la pagina
    if (InputValue.trim().length > 2) {
      setCategories(cats => [ InputValue, ...cats, ]);//Utilizamos la propiedad setCategories y añadimos el valor del input a la lista
      setInputValue('');
      
    }
  }
  return (
    <form onSubmit={handleSubmit}>
        <input 
          type="text"
          value={InputValue}
          onChange={handleInput}
        />
    </form>
  )
}
//Hacemos que la propiedad setCategorie sea obligatoria
AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}
