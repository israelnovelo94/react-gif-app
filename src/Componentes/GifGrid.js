import React from 'react'
import { useFetchGif } from '../hooks/useFetchGif';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

    // Llamamos al custom hook y hacemos la desestructuracion de los datos
    const {data,loading} = useFetchGif(category);

  return (
    <>
        <h3>{category}</h3>
        {/* Hacemos una validacion para desplegar el loading en lo que carga la información */}
        {loading && <h4>Loading...</h4>}
        <div className='card-grid'>
            
            { 
                //mapeamos o recorremos la data para sacar la informacion del arreglo
                data.map( img => 
                  
                    <GifGridItem 
                    key={img.id}
                    {...img}
                    />
                    ) 
            }
            

        </div> 
    </>
  )
}
