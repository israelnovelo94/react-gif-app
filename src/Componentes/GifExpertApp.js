import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

const GifExpertApp = ( {defaultCategories = []}) => {
    

    const [categories, setCategories] = useState(defaultCategories);

    // const handleAdd = () => {
    //     setCategories( cats => [...cats, 'HunterxHunter']);
    // }
    return (
        <>
        <h2>Gix Expert App</h2>
        <hr />

        <AddCategory setCategories={ setCategories } />
        <ol>
        {
            categories.map( (category) => 

                    <GifGrid 
                        key={ category }
                        category={ category }
                    />
            )
        }
        </ol>
        </>
        
        
    )
}

export default GifExpertApp;