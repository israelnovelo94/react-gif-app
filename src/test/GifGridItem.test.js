import { shallow } from "enzyme"
import { GifGridItem } from "../Componentes/GifGridItem"

describe('Probando el GifGridItem', () => {
    
    const title = 'Titulo';
    const url   = 'www.google.com';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('Prueba de snapshot del componente', () => {


        expect( wrapper ).toMatchSnapshot();
    });

    test('Verificando que llegue el titulo', () => {
        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(title);
    });

    test('Verificando los atributos de la imagen', () => {
        const img = wrapper.find('img');
        // console.log(img.html());

        expect(img.prop('src')).toBe(url);
        expect(img.props().alt).toBe(title);
    });

    test('Verificar que tenga la clase animate__fadeInDown', () => {
        const div = wrapper.find('div');

        // console.log(div.hasClass('animate__fadeInDown'));

        expect(div.hasClass('animate__fadeInDown')).toBe( true )
    })
}) 