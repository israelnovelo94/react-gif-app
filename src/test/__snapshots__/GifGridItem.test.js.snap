// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el GifGridItem Prueba de snapshot del componente 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="Titulo"
    src="www.google.com"
  />
  <p>
    Titulo
  </p>
</div>
`;
